testbed/integration/data_providers.go

```diff
@@ -0,0 +1,49 @@
+package integration
+
+import (
+	"sync/atomic"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/testbed/testbed"
+	"go.opentelemetry.io/collector/pdata/plog"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+// sampleConfigsDataProvider in an implementation of the DataProvider
```

Comment:
```suggestion
var _ testbed.DataProvider = &sampleConfigsDataProvider{}

```

I forget what these statements are called, but I would just verify that the type implements the interface like this. This also allows you to remove the comments on the methods since you can see which methods the interface provides.